config/confighttp/confighttp.go

```diff
@@ -46,3 +48,34 @@ func (hcs *HTTPClientSettings) ToClient() (*http.Client, error) {
 		Timeout:   hcs.Timeout,
 	}, nil
 }
+
+type HTTPServerSettings struct {
+	// Endpoint configures the listening address for the server.
+	Endpoint string `mapstructure:"endpoint"`
+
+	// TLSSetting struct exposes TLS client configuration.
+	TLSSetting *configtls.TLSServerSetting `mapstructure:"tls_settings, omitempty"`
+}
+
+func (hss *HTTPServerSettings) ToListener() (net.Listener, error) {
+	listener, err := net.Listen("tcp", hss.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if hss.TLSSetting != nil {
+		var tlsCfg *tls.Config
+		tlsCfg, err = hss.TLSSetting.LoadTLSConfig()
+		if err != nil {
+			return nil, err
+		}
+		listener = tls.NewListener(listener, tlsCfg)
+	}
+	return listener, nil
+}
+
+func (hss *HTTPServerSettings) ToServer(handler http.Handler) *http.Server {
```

Comment:
Is there any point in having this function on the settings struct? Seems like it doesn't do anything related to the settings

Reply:
Will add more settings like timeout that will need to be on the server, and stuff and want users to start using this